app/lib/features/onboarding/pages/intro_page.dart

```diff
@@ -11,11 +12,17 @@ class IntroPage extends StatefulWidget {
 }
 
 class _IntroPageState extends State<IntroPage> {
-  final List<String> introTexts = [
-    'Simple to use.',
-    'Secure with E2E Encryption.',
-    'Powerful for any organizer.',
-  ];
+  List<String> introTexts = [];
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    introTexts = [
+      L10n.of(context).simpleToUse,
+      L10n.of(context).secureWithE2EEncryption,
+      L10n.of(context).powerfulForAnyOrganizer,
+    ];
+  }
```

Comment:
why is this a state rather than just a let in `build`?

Reply:
I need to initialise `introTexts` array data but I can't directly do it at `variable declaration` time or in `initState` as for localise string, I need `context`.

So there were two options I had where I can do this `initialisation` with `context`.
1) `didChangeDependencies`
2) `build` method 

I preferred to do that on `didChangeDependencies` as just called right after `initState` and before `build` in lifecycle events. And our purpose it also kind of initialisation of static string values.

Reference:
https://api.flutter.dev/flutter/widgets/State/didChangeDependencies.html
![Screenshot 2024-03-15 at 1 05 10 PM](https://github.com/acterglobal/a3/assets/51526480/3a7511b7-163d-4153-a7dc-a083d9f45915)

Reply:
the reason I dislike doing that in `didChangeDependencies` rather than in `build` is that the function is there to initialize _state_. But this isn't state. This is _technically_ static, never changing content. Thus it should not be acted upon as if it was ever changing - it's just confusing to make that a _variable_ where it being `final` (by just using it in `build`) would do.